infra/terraform/global/s3.tf

```diff
@@ -15,3 +15,27 @@ resource "aws_s3_bucket" "terraform_state" {
         enabled = true
     }
 }
+
+resource "aws_kms_key" "s3_uploads" {
+    description = "Key for upload object encryption"
+    deletion_window_in_days = 10
```

Comment:
I read the terraform documentation on this `deletion_window_in_days` argument.

> Duration in days after which the key is deleted after destruction of the resource, must be between 7 and 30 days. [...]

So we may delete the terraform resource but the KMS key sticks around for another 10 days in AWS? Any reasons why it's not deleted straight away? (maybe because if you accidentally delete it you will "never" be able to decrypt your data anymore? so they keep it around just-in-case?)

Reply:
Yep, exactly - you can't delete a KMS key immediately, and 7 days in the minimum.